controllers/dockercluster_controller.go

```diff
@@ -39,84 +48,123 @@ type DockerClusterReconciler struct {
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters/status,verbs=get;update;patch
 
-// Reconcile handles DockerCluster events
-func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr error) {
+// Reconcile reads that state of the cluster for a DockerCluster object and makes changes based on the state read
+// and what is in the DockerCluster.Spec
+func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
 	ctx := context.Background()
-	log := r.Log.WithValues("dockercluster", req.NamespacedName)
-	log.Info("Reconciling cluster")
+	log := log.Log.WithName(controllerName).WithValues("dockerCluster", req.NamespacedName)
 
-	dockerCluster := &infrastructurev1alpha2.DockerCluster{}
+	// Fetch the DockerCluster instance
+	dockerCluster := &infrav1.DockerCluster{}
 	if err := r.Client.Get(ctx, req.NamespacedName, dockerCluster); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.Error(err, "failed to get DockerCluster")
 		return ctrl.Result{}, err
 	}
 
+	// Fetch the Cluster.
 	cluster, err := util.GetOwnerCluster(ctx, r.Client, dockerCluster.ObjectMeta)
 	if err != nil {
-		log.Error(err, "Failed to get owning cluster")
-		return ctrl.Result{}, err
+		return reconcile.Result{}, err
 	}
 	if cluster == nil {
-		log.Error(err, "Waiting for an OwnerReference to appear")
-		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+		log.Info("Waiting for Cluster Controller to set OwnerRef on DockerCluster")
+		return reconcile.Result{}, nil
+	}
+
+	log = log.WithValues("cluster", cluster.Name)
+
+	// Create a service for managing the docker loadbalancer.
+	lbsvc, err := docker.NewLoadBalancer(cluster.Name, log)
+	if err != nil {
+		return reconcile.Result{}, errors.Errorf("failed to create scope: %+v", err)
 	}
 
-	// Store Config's state, pre-modifications, to allow patching
-	patchCluster := client.MergeFrom(dockerCluster.DeepCopy())
+	// Store dockerCluster's state, pre-modifications, to allow patching
+	dockerClusterPatch := client.MergeFrom(dockerCluster.DeepCopy())
```

Comment:
I have a PR out to switch to using the patch helper from cluster-api here: https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/202